src/briefcase/platforms/web/static.py

```diff
@@ -96,33 +116,333 @@ def _trim_file(self, path, sentinel):
+        try:
+            file_text = target_path.read_text(encoding="utf-8")
+        except FileNotFoundError:
+            self.console.warning(f"  Target {filename} not found; skipping inserts.")
+            return
+
+        # Each insert slot and its package contributions are processed in sorted order
+        for insert, pkg_contribs in sorted(inserts.items()):
+            # Build bodies from the same contributions
+            html_body = "\n".join(
+                HTML_BANNER.format(package=pkg, content=text)
+                for pkg, text in sorted(pkg_contribs.items())
+                if text
+            )
+            css_body = "\n".join(
+                CSS_BANNER.format(package=pkg, content=text)
+                for pkg, text in sorted(pkg_contribs.items())
+                if text
+            )
+            body_map = {"html": html_body, "css": css_body}
+
+            slot = re.escape(insert)
+
+            # Build the compiled patterns directly once per slot
+            compiled_markers = [
+                (
+                    re.compile(
+                        rf"(^[ \t]*)<!--@@ {slot}:start @@-->.*?<!--@@ {slot}:end @@-->",
+                        flags=re.MULTILINE | re.DOTALL,
+                    ),
+                    r"{indent}<!--@@ {insert}:start @@-->\n{content}{indent}<!--@@ {insert}:end @@-->",
+                    "html",
+                ),
+                (
+                    re.compile(
+                        rf"(^[ \t]*)/\*@@ {slot}:start @@\*/.*?/\*@@ {slot}:end @@\*/",
+                        flags=re.MULTILINE | re.DOTALL,
+                    ),
+                    r"{indent}/*@@ {insert}:start @@*/\n{content}{indent}/*@@ {insert}:end @@*/",
+                    "css",
+                ),
+            ]
+
+            # Apply all matching marker styles
+            any_match = False
+            for pattern, repl_tmpl, kind in compiled_markers:
+                # Search for pattern within file.
+                pattern_found = pattern.search(file_text)
+                if pattern_found:
+                    # Indent content to align with markers.
+                    indented_content = indent(
+                        body_map.get(kind, ""), pattern_found.group(1)
+                    )
+                    file_text = pattern.sub(
+                        repl_tmpl.format(
+                            indent=pattern_found.group(1),
+                            insert=insert,
+                            content=indented_content,
+                        ),
+                        file_text,
+                    )
+                    any_match = True
+
+            if not any_match:
+                self.console.warning(
+                    f"  Slot '{insert}' markers not found in {filename}; skipping."
+                )
+
+        # Save modified content
+        target_path.write_text(file_text, encoding="utf-8")
+
+    def _write_pyscript_insert(
```

Comment:
Minor housekeeping issue - this is no longer writing anything, so the method name and docstring is misleading.